rustls/src/crypto/ring/kx.rs

```diff
@@ -102,11 +133,16 @@ struct KeyExchange {
     agreement_algorithm: &'static agreement::Algorithm,
     priv_key: agreement::EphemeralPrivateKey,
     pub_key: agreement::PublicKey,
+    pub_key_validator: fn(&[u8]) -> bool,
 }
 
 impl ActiveKeyExchange for KeyExchange {
     /// Completes the key exchange, given the peer's public key.
     fn complete(self: Box<Self>, peer: &[u8]) -> Result<SharedSecret, Error> {
+        match (self.pub_key_validator)(peer) {
+            true => {}
+            false => return Err(PeerMisbehaved::InvalidKeyShare.into()),
+        };
```

Comment:
Did you think the `match` with the empty `true` arm was more readable than an `if !(self.pub_key_validator)(peer) { return Err(PeerMisbehaved::InvalidKeyShare.into()); }` ? The latter seems a bit more concise but I don't feel strongly.